spec/models/user_spec.rb

```diff
@@ -13,75 +13,95 @@
       expect(assc.macro).to eq :has_one
     end
 
-    it 'has many friend requests' do
-      assc = described_class.reflect_on_association(:requests)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'has many pending_friends' do
-      assc = described_class.reflect_on_association(:pending_friends)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'is expected to destroy dependent requests' do
-      request
-      expect { request.receiver.destroy }.to change { Request.count }.by(-1)
-    end
-
-    it 'has many friendships' do
-      assc = described_class.reflect_on_association(:friendships)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'returns friendships when called upon friendships method' do
-      expect(friendship.user.friendships.first).to eq(friendship) 
-    end
-
-    it 'is expected to destroy dependent friendships' do
-      friendship
-      expect { friendship.user.destroy }.to change { Friendship.count }.by(-1)
-    end
-
-    it 'has many friends' do
-      assc = described_class.reflect_on_association(:friends)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'has many posts' do
-      assc = described_class.reflect_on_association(:posts)
-      expect(assc.macro).to eq :has_many
+    context 'Friend Requests' do
```

Comment:
I recommend the following organization for your specs, example here is for `user_spec.rb` but you may use the ff template for your other model specs:

```ruby
RSpec.describe User do
  # test data declarations here

  describe "validations" do
    context "complete user details" do
      it "is valid" do
        # expectations here
     end
     context "name missing" do
        it "is invalid" do
         # expectations here
        end
     end
    end
  end

  describe "associations" do
    context "user's association w/ profile" do
      it ...
    end
  end
  describe "instance methods" do
   # contexts here...
  end
end
```